Realm/Realm/DatabaseTypes/InvalidRealmObject.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ////////////////////////////////////////////////////////////////////////////
+// //
+// // Copyright 2021 Realm Inc.
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License")
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// // http://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+// //
+// ////////////////////////////////////////////////////////////////////////////
+
+namespace Realms
+{
+    /// <summary>
+    /// Special invalid realm object that is used to avoid an exception in WPF
+    /// when deleting an element from a collection bound to UI (<see href="https://github.com/realm/realm-dotnet/issues/1903">#1903</see>).
+    /// </summary>
+    internal class InvalidRealmObject : RealmObjectBase
+    {
+        private static readonly InvalidRealmObject _instance = new InvalidRealmObject();
+
+        private InvalidRealmObject() { }
+
+        public static InvalidRealmObject Instance => _instance;
+
+        // The method is overriden to avoid the bug in WPF
+        public override bool Equals(object obj)
+        {
+            return true;
+        }
```

Comment:
Let's be a bit more restrictive with the equality check here:

```suggestion
        public override bool Equals(object obj) => obj is RealmObjectBase robj && !robj.IsValid;
```

Reply:
Regarding this, I was trying to be less restrictive because we use this in `RealmCollectionBase<T>`, and we don't know if `T` is actually a `RealmObjectBase`. Probably, because of this, we don't even need to have this fake object derive from RealmObjectBase and just use a completely independent filler object. What do you think?

Reply:
Ugh... I see. Yes, then let's make it a completely independent object. That being said, we should probably check customize `Equals` on `RealmObjectBase` and make sure to return `true` if `other` is `InvalidRealmObject` and `this.IsValid == false`.

Reply:
I have just pushed a commit that makes this a separate object. Regarding Equals... I understand the sentiment, but do you think we should go the extra mile for something that is a patch for a bug in WPF and has a very limited scope?

Reply:
I guess the issue is that there might be other controls out there that do the inverse check - i.e. `original.Equals(deleted)` which will return `false`. While there's not much we can do about non-object types, we can still make objects behave as expected.